Replications not support for Windows image with foreign layers

Replications not support for Windows image with foreign layers, in replication job pull image from source image will failed.

Here is the job logs.
```
2019-02-21T15:50:23Z [INFO] initialization completed: repository: windows/hello-world, tags: [latest], source registry: URL-http://harbor-17x-harbor-core insecure-false, destination registry: URL-https://harbor.10.92.0.132.xip.io insecure-true
2019-02-21T15:50:23Z [INFO] project windows is created on destination registry
2019-02-21T15:50:23Z [INFO] manifest of windows/hello-world:latest pulled successfully from source registry: sha256:4c0c09dd5a7fe632acd637acf0a676ead5ce2aeaa5b79c181636171bf57aa153
2019-02-21T15:50:23Z [INFO] transferring blob sha256:365a278e3083dfdb08b9a2251d4cda5fe9261e9a0345507279a64244085c6060 of windows/hello-world:latest to the destination registry ...
2019-02-21T15:50:23Z [INFO] blob sha256:365a278e3083dfdb08b9a2251d4cda5fe9261e9a0345507279a64244085c6060 of windows/hello-world:latest transferred to the destination registry completed
2019-02-21T15:50:23Z [INFO] transferring blob sha256:407ada6e90de9752a53cb9f52b7947a0e38a9b21a349970ace15c68890d72511 of windows/hello-world:latest to the destination registry ...
2019-02-21T15:50:23Z [ERROR] [transfer.go:291]: an error occurred while pulling blob sha256:407ada6e90de9752a53cb9f52b7947a0e38a9b21a349970ace15c68890d72511 of windows/hello-world:latest from the source registry: 404 {"errors":[{"code":"BLOB_UNKNOWN","message":"blob unknown to registry","detail":"sha256:407ada6e90de9752a53cb9f52b7947a0e38a9b21a349970ace15c68890d72511"}]}
```

--- a/src/jobservice/job/impl/replication/transfer.go
+++ b/src/jobservice/job/impl/replication/transfer.go
@@ -272,6 +272,12 @@ func (t *Transfer) transferLayers(tag string, blobs []distribution.Descriptor) e
 		}
 
 		digest := blob.Digest.String()
+
+		if blob.MediaType == schema2.MediaTypeForeignLayer {
+			t.logger.Infof("blob %s of %s:%s is an foreign layer, skip", digest, repository, tag)
+			continue
+		}
+
 		exist, err := t.dstRegistry.BlobExist(digest)
 		if err != nil {
 			t.logger.Errorf("an error occurred while checking existence of blob %s of %s:%s on destination registry: %v",

